### PR TITLE
update golang to 1.18 and k8s to 0.24.2

### PR DESCRIPTION
**Pull request by ranakan19**
update the following:
1. golang to 1.18
2. k8s to 0.24.2
3. golangcli to v3

### CODE REVIEW

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -39,18 +39,17 @@ jobs:
...
         
     - name: Checkout code
       uses: actions/checkout@v2
     
     - name: Lint
-      uses: golangci/golangci-lint-action@v2
+      uses: golangci/golangci-lint-action@v3
       with:
-        version: v1.43.0
-        skip-go-installation: true # we already installed Go with `actions/setup-go@v2` above
```
Comment (xcoulon) [inline .github/workflows/ci-build.yml:53]: was removing this `skip-go-installation` intentional?

File: pkg/proxy/proxy.go [hunk 2]
```diff
@@ -76,7 +76,7 @@ func (p *Proxy) StartProxy() *http.Server {
 
 	// listen concurrently to allow for graceful shutdown
 	log.Info(nil, "Starting the Proxy server...")
-	srv := &http.Server{Addr: ":" + ProxyPort, Handler: handler}
+	srv := &http.Server{Addr: ":" + ProxyPort, Handler: handler, ReadHeaderTimeout: 2 * time.Second}
```
Comment (xcoulon) [inline pkg/proxy/proxy.go:79]: did you have any test failures that got resolved with this setting?

**Review by xcoulon — APPROVED** [review]
approved with minor non-blocking questions

File: pkg/proxy/proxy.go (hunk 2 above)
Comment (ranakan19) [reply pkg/proxy/proxy.go:79]: yes, make lint was giving the following error, which got resolved:
``` pkg/proxy/proxy.go:79: File is not `gofmt`-ed with `-s` (gofmt)
        srv := &http.Server{Addr: ":" + ProxyPort, Handler: handler, ReadHeaderTimeout: 2*time.Second}```

File: .github/workflows/ci-build.yml (hunk 1 above)
Comment (ranakan19) [reply .github/workflows/ci-build.yml:53]: yes, it was not needed with  golangci/golangci-lint-action@v3

**Review by mfrancisc — APPROVED** [review]

**Review by MatousJobanek — APPROVED** [review]

**Pull request merged**